elasticsearch-transport/lib/elasticsearch/transport/transport/base.rb

```diff
@@ -42,7 +42,8 @@ def initialize(arguments={}, &block)
           @logger      = options[:logger]
           @tracer      = options[:tracer]
 
-          @sniffer     = options[:sniffer_class] ? options[:sniffer_class].new(self) : Sniffer.new(self)
+          sniffer_class = options[:sniffer_class] ? options[:sniffer_class] : Sniffer
+          @sniffer     = sniffer_class.new(self, logger)
```

Comment:
I must say I'm also a fan of using @var whenever possible, makes the code easier to read